packages/rx_bloc_cli/mason_templates/bricks/rx_bloc_base/__brick__/test/feature_counter/views/helpers/counter_helper.dart

```diff
@@ -0,0 +1,68 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:golden_toolkit/golden_toolkit.dart';
+import 'package:provider/provider.dart';
+
+import 'package:{{project_name}}/feature_counter/blocs/counter_bloc.dart';
+import 'package:{{project_name}}/feature_counter/views/counter_page.dart';
+
+import '../../../helpers/golden_helper.dart';
+import '../../../helpers/models/scenario.dart';
+import '../../mocks/counter_bloc_mock.dart';
+
+typedef CustomDeviceBuilder = DeviceBuilder Function(WidgetTester);
+
+// these will each get generated as separate golden master files
+List<CustomDeviceBuilder> customDeviceBuilders = [
+  deviceBuilderWithCounterScenarios,
+  deviceBuilderWithErrorScenario,
+  deviceBuilderWithLoadingScenario,
+];
+
+//test each counter state
+DeviceBuilder deviceBuilderWithCounterScenarios(WidgetTester tester) =>
+//return a generated DeviceBuilder with specified widget and list of scenarios
+    generateDeviceBuilder(
+      counterPageFactory(count: 2),
+      scenarios: [
+        Scenario(name: 'Default'),
+      ],
+    );
+
+//test error snackbar scenario
+DeviceBuilder deviceBuilderWithErrorScenario(WidgetTester tester) =>

```

Comment:
From what I have observed, most developers look at the actual test to see all tested scenarios (in our case this is `counter_page_test.dart`). For reference check `counter_bloc_test.dart`

With that being said @GoranPrime do you think you can:

- refactor  the `counter_page_test.dart` in a way that inside that test we can see all tested scenarios (it should be as small/simple as possible) 
- Make all helpers/utils as generic as possible, so that they can be reused across all tests

Reply:
Great suggestions. I've refactored `counter_page_test.dart` and `golden_helper.dart` to be more flexible, and removed `counter_helper.dart` entirely